core/embed/boardloader/main.c

```diff
@@ -280,6 +285,23 @@ int main(void) {
 #if defined USE_SD_CARD
   sdcard_init();
 
+#ifdef STM32U5
+  secret_ensure_initialized();
+
+  const image_header *old_hdr = read_image_header(
```

Comment:
please add a comment here that this is intro for the sdcard bootloader install, and why

and below, after the "check/copy" sequence, that we're (possibly) reading the bootloader again and this time it's the changed one

Reply:
added https://github.com/trezor/trezor-firmware/commit/c8d2a20f2e4b9e9415b74a69eb4ae0d9c22ec207